server/neptune/workflows/internal/activities/execute.go

```diff
@@ -1,12 +1,90 @@
 package activities
 
-import "context"
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
 
-type executeCommandActivities struct{}
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/runtime/cache"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/steps"
+)
+
+type executeCommandActivities struct {
+	versionCache cache.ExecutionVersionCache
+
+	DefaultTFVersion *version.Version
+
+	// TerraformBinDir is the directory where Atlantis downloads Terraform binaries.
+	TerraformBinDir string
+}
+
+type ExecuteCommandResponse struct {
+	Output string
+	Error  error
+}
 
 type ExecuteCommandRequest struct {
+	Step          steps.Step
+	CustomEnvVars map[string]string
+	Envs          map[string]string
+
+	Path string
+	// TerraformVersion is the version of terraform we should use when executing
+	// commands for this project. This can be set to nil in which case we will
+	// use the default Atlantis terraform version.
+	TerraformVersion *version.Version
 }
 
-func (t *terraformActivities) ExecuteCommand(ctx context.Context, request ExecuteCommandRequest) error {
-	return nil
+func (t *executeCommandActivities) ExecuteCommand(ctx context.Context, request ExecuteCommandRequest) ExecuteCommandResponse {
+
+	terraformVersion := t.DefaultTFVersion
+	if request.TerraformVersion != nil {
+		terraformVersion = request.TerraformVersion
+	}
+
+	// Ensure version exists
+	_, err := t.versionCache.Get(terraformVersion)
+	if err != nil {
+		return ExecuteCommandResponse{
+			Output: "",
+			Error:  errors.Wrapf(err, "getting version %s", request.TerraformVersion),
+		}
+	}
+
+	terraformEnvVars := map[string]string{
+		"ATLANTIS_TERRAFORM_VERSION": terraformVersion.String(),
+		"PATH":                       fmt.Sprintf("%s:%s", os.Getenv("PATH"), t.TerraformBinDir),
+	}
+
+	cmd := exec.Command("sh", "-c", request.Step.RunCommand) // #nosec
```

Comment:
What about log streaming?

Reply:
From my understanding, we don't support streaming logs for the run steps right? If we want to, I can tackle this in separate task to keep this PR short.